mypy/checker.py

```diff
@@ -1064,30 +1064,33 @@ def check_inplace_operator_method(self, defn: FuncBase) -> None:
 
     def check_getattr_method(self, typ: CallableType, context: Context, name: str) -> None:
         if len(self.scope.stack) == 1:
-            # module-level __getattr__
+            # module scope
             if name == '__getattribute__':
                 self.msg.fail('__getattribute__ is not valid at the module level', context)
                 return
-            elif name == '__getattr__' and not self.is_stub:
-                self.msg.fail('__getattr__ is not valid at the module level outside a stub file',
-                              context)
-                return
+            # __getattr__ is fine at the module level as of Python 3.7 (PEP 562). We could
```

Comment:
Hm, I think this could actually bite people more than you'd think. People already get confused about why `__getattr__` works in stubs and not in runtime modules without mypy allowing it in non-stubs. Also code that supports both 3.7 and below probably wouldn't be able to use `__getattr__`, I can't think of a non-hacky way in which it would work at runtime...

Reply:
I think it is actually fine. We don't warn about `__setattr__` being unsupported etc.